folly/container/test/tape_bench.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/container/tape.h>
+
+#include <random>
+#include <string>
+#include <vector>
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+
+namespace {
+
+using vec_str = std::vector<std::string>;
+using st_tape = folly::string_tape;
+using vv_int = std::vector<std::vector<int>>;
+using tv_int = folly::tape<std::vector<int>>;
+
+template <typename Cont>
+struct ContGenerator {
+  ContGenerator(std::size_t minLen, std::size_t maxLen)
+      : g(minLen + maxLen), len(minLen, maxLen) {}
+
+  const Cont& operator()() {
+    buf.resize(len(g));
```

Comment:
@DenisYaroshevskiy shouldn't this be `buf.reserve(len(g));` instead?

`resize` + `emplace_back()` will increase the size. While `reserve` will reserve the space and array won't grow dynamically as often.